### PR TITLE
partOf should contain nested ancestors

### PR DESCRIPTION
**Pull request by warrd**
## Issue

https://github.com/wellcomecollection/platform/issues/4671

## Description

The `partOf` field currently incorrectly contains an array of all ancestors, but should contain only direct parents, with early ancestors nested within.

Here we allow relation fields to be nested, currently with only `partOf` containing nesting.

NOTE:  For backwards compatibility we include the unnested ancestors ordered from root to parent, with the parent additionally containing the nested data. This will be removed once the front end is switched over.

### CODE REVIEW

File: api/api/src/main/scala/uk/ac/wellcome/platform/api/services/RelatedWorkService.scala [hunk 1]
```diff
@@ -31,7 +24,7 @@ class RelatedWorkService(elasticsearchService: ElasticsearchService)(
   def retrieveRelatedWorks(index: Index,
                            work: IdentifiedWork): Future[Result[RelatedWorks]] =
     work.data.collectionPath match {
-      case None => Future.successful(Right(RelatedWorks(Nil, Nil, Nil, Nil)))
+      case None => Future.successful(Right(RelatedWorks(Some(Nil), Some(Nil), Some(Nil), Some(Nil))))
```
Comment (jamieparkinson) [inline api/api/src/main/scala/uk/ac/wellcome/platform/api/services/RelatedWorkService.scala]: Could be worth adding a `def empty` for this case to the companion object as it's used in a couple of places / looks nice

File: common/display/src/main/scala/uk/ac/wellcome/display/models/DisplayWork.scala [hunk 2]
```diff
@@ -187,4 +187,40 @@ case object DisplayWork {
 
   def apply(work: IdentifiedWork): DisplayWork =
     DisplayWork(work = work, includes = WorksIncludes())
+
+  def apply(work: IdentifiedWork, includes: WorksIncludes, relatedWorks: RelatedWorks): DisplayWork =
```
Comment (jamieparkinson) [inline common/display/src/main/scala/uk/ac/wellcome/display/models/DisplayWork.scala]: :+1: a better place for this to live imo

File: api/api/src/main/scala/uk/ac/wellcome/platform/api/services/RelatedWorkService.scala (hunk 1 above)
Comment (warrd) [reply api/api/src/main/scala/uk/ac/wellcome/platform/api/services/RelatedWorkService.scala]: There is already a `def none` for when its `None`, this is `Some(Nil)` :see_no_evil: .. This means there actually no related works, the other means we don't actually know at that point, maybe could be `unknown` / `none` or something?

Comment (jamieparkinson) [reply api/api/src/main/scala/uk/ac/wellcome/platform/api/services/RelatedWorkService.scala]: Yeah the naming is tricky... how about `RelatedWorks.nil`?

Comment (warrd) [reply api/api/src/main/scala/uk/ac/wellcome/platform/api/services/RelatedWorkService.scala]: :+1: I like `RelatedWorks.nil` and `RelatedWorks.unknown`

**Review by kenoir — APPROVED** [review]

**Pull request merged**